### ISSUE TITLE
_gui_input is not propagated to parent for keyboard and joy inputs

### ISSUE THREAD

**Issue by Jules5** [godotengine/godot#98682]
### Tested versions

- Tested in 4.3.stable

### System information

Godot v4.3.stable - Windows 10.0.22631 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1060 with Max-Q Design (NVIDIA; 31.0.15.5123) - Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz (12 Threads)

### Issue description

_gui_input is only called on the focused control, and is not propagated to parent even if accept_event is NOT called.
Note that is does not affect mouse events (when the mouse_filter of the child is set to Pass).

### Steps to reproduce

- Launch the MRP, th button on screen will be automatically focused
- Press space, notice in the output that _gui_input is only called on the child.
- Notice that if you move the mouse on the button or even click, _gui_input is called on both the child and the parent.

### Minimal reproduction project (MRP)

[bug-gui-input-propagation.zip](https://github.com/user-attachments/files/17579412/bug-gui-input-propagation.zip)

**Comment by Jules5** [comment]
I found a PR that fixes this issue: #81499.
It seems it has been waived because there was no use case for this. I added a [comment](https://github.com/godotengine/godot/pull/81499#issuecomment-2449769076) with a description of mine, which in my opinion is very valid (keyboard/gamepad compatible GUI).

**Comment by Jules5** [comment]
See this [comment](https://github.com/godotengine/godot/pull/81499#issuecomment-2449769076) describing why this issue is cumbersome and what this [PR](https://github.com/godotengine/godot/pull/81499) would improve.

**Comment by 3lis4** [comment]
I completely agree with @Jules5, the current behavior forces us to do cumbersome and convoluted code to make a not-that-complex menu navigable with gamepad/keyboard. It would be much more intuitive to write/read code, and similar to how mouse inputs are handled, with what @Jules5  describes.
@Sauermann said in #81499 we needed use cases to know if _gui_input propagation for keyboard/gamepad inputs would be useful or not. @Jules5 and I gave use cases, [see mine here](https://github.com/godotengine/godot/pull/81499#issuecomment-2034625861). With these use cases in mind, it seems now that it would be better to change the behavior #81499 over modifying the documentation #7899 (which was preferred at the time). Is there any argument against this ?
